src/mip/mip_logging.c

```diff
@@ -0,0 +1,79 @@
+
+#include "mip_logging.h"
+
+#include <stddef.h>
+
+////////////////////////////////////////////////////////////////////////////////
+///@brief Global logging callback. Do not access directly
+mip_log_callback _mip_log_callback = NULL;
+
+////////////////////////////////////////////////////////////////////////////////
+///@brief Global logging level. Do not access directly
+mip_log_level _mip_log_level = MIP_LOG_LEVEL_OFF;
+
+////////////////////////////////////////////////////////////////////////////////
+///@brief Global logging user data. Do not access directly
+void* _mip_log_user_data = NULL;
+
+////////////////////////////////////////////////////////////////////////////////
+///@brief Initializes the logger with a callback and user data.
+///       Call MIP_LOG_INIT instead of using this function directly
+///
+///@param callback The callback to execute when there is data to log
+///@param level    The level that the MIP SDK should log at
```

Comment:
Maybe add a note here saying that this doesn't HAVE to be called, unless the user wants logging